lib/Sema/TypeCheckDecl.cpp

```diff
@@ -868,6 +868,12 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
   if (!currentFile || currentDC->isLocalContext())
     return;
 
+  // Cannot define a type called 'Any'
+  if (current->getFullName() == tc.Context.Id_Any) {
```

Comment:
I had to enforce this because of the way I handle an UnresolvedDeclRefExpr: the [current implementation](https://github.com/apple/swift/pull/3293/files#diff-e38c4e32f8ef5d6750d4c6a27b6b5397R445) returns the found Any type before the unqualified lookup, meaning that a type `Any` declared by the user would never get found.

I could change it to search for this `Any` keyword *after* the unqualified lookup, but then if the user wanted to access the protocol<> meaning of Any, they would have to qualify a Swift keyword like `Swift.Any` which has no precedent in the language. What do you think?